File: chrome/browser/share/android/java/src/org/chromium/chrome/browser/share/link_to_text/LinkToTextMetricsBridge.java
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.share.link_to_text;

import org.chromium.base.annotations.NativeMethods;

/**
 * A Java API for recording Link to Text metrics.
 */
public class LinkToTextMetricsBridge {
    public static void logGenerateErrorTabHidden() {
        LinkToTextMetricsBridgeJni.get().logGenerateErrorTabHidden();
    }

    public static void logGenerateErrorOmniboxNavigation() {
        LinkToTextMetricsBridgeJni.get().logGenerateErrorOmniboxNavigation();
    }

    public static void logGenerateErrorTabCrash() {
        LinkToTextMetricsBridgeJni.get().logGenerateErrorTabCrash();
    }

    public static void logGenerateErrorIFrame() {
        LinkToTextMetricsBridgeJni.get().logGenerateErrorIFrame();
    }

    @NativeMethods
    interface Natives {
        void logGenerateErrorTabHidden();
        void logGenerateErrorOmniboxNavigation();
        void logGenerateErrorTabCrash();
        void logGenerateErrorIFrame();
    }
}

File: chrome/browser/share/link_to_text_metrics_bridge.cc
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <jni.h>

#include "chrome/browser/share/android/jni_headers/LinkToTextMetricsBridge_jni.h"
#include "components/shared_highlighting/core/common/shared_highlighting_metrics.h"

static void JNI_LinkToTextMetricsBridge_LogGenerateErrorTabHidden(JNIEnv* env) {
  shared_highlighting::LogGenerateErrorTabHidden();
}

static void JNI_LinkToTextMetricsBridge_LogGenerateErrorOmniboxNavigation(
    JNIEnv* env) {
  shared_highlighting::LogGenerateErrorOmniboxNavigation();
}

static void JNI_LinkToTextMetricsBridge_LogGenerateErrorTabCrash(JNIEnv* env) {
  shared_highlighting::LogGenerateErrorTabCrash();
}

static void JNI_LinkToTextMetricsBridge_LogGenerateErrorIFrame(JNIEnv* env) {
  shared_highlighting::LogGenerateErrorIFrame();
}
